### ISSUE TITLE
Compilation error on Slackware Linux with gcc 14

### ISSUE THREAD

**Issue by xbasics** [duckdb/duckdb-r#193]
Tried compiling v1.0.0 within R environment as part of installation and getting the following error.  Installed the latest version of the re2 package without any issue, yet there seems to be a problem with the one included.

duckdb/third_party/re2/re2/prog.cc: In member function ‘const void* duckdb_re2::Prog::PrefixAccel_FrontAndBack(const void*, size_t)’:
duckdb/third_party/re2/re2/prog.cc:1146:45: error: ‘prefix_front_’ was not declared in this scope
 1146 |     const __m256i f_set1 = _mm256_set1_epi8(prefix_front_);
      |                                             ^~~~~~~~~~~~~
duckdb/third_party/re2/re2/prog.cc:1147:45: error: ‘prefix_back_’ was not declared in this scope; did you mean ‘prefix_size_’?
 1147 |     const __m256i b_set1 = _mm256_set1_epi8(prefix_back_);
      |                                             ^~~~~~~~~~~~
      |                                             prefix_size_
make: *** [/usr/lib64/R/etc/Makeconf:200: duckdb/third_party/re2/re2/prog.o] Error 1
ERROR: compilation failed for package ‘duckdb’
* removing ‘~/.R_LIBS/duckdb’
* restoring previous ‘~/.R_LIBS/duckdb’
Warning in install.packages :
  installation of package ‘duckdb’ had non-zero exit status

The downloaded source packages are in
	‘/tmp/RtmpQngRjl/downloaded_packages’

**Comment by krlmlr** [comment]
Thank you. What OS and version is this, what compiler are you using? Can you compile the duckdb sources at https://github.com/duckdb/duckdb ?

**Comment by xbasics** [comment]
Thanks for your reply!  I'm running Linux (Slackware current) using GCC 14.1.0 w/ GLibc 2.39 on an AMD Ryzen 7 4800H processor.  In terms of R environment, I'm using R 4.4.1.  I compile R on my local machine using the following C/CXX flags, which are kept was part of the subsequent package installations.

-O3 -fPIC -march=native -pipe

**Comment by xbasics** [comment]
Thanks for your reply!  I'm running Linux (Slackware current) using GCC 14.1.0 w/ GLibc 2.39 on an AMD Ryzen 7 4800H processor.  In terms of R environment, I'm using R 4.4.1.  I compile R on my local machine using the following C/CXX flags, which are kept was part of the subsequent package installations.

-O3 -fPIC -march=native -pipe

In terms of compiling duckdb-r sources by themselves, I did clone the (current) main branch and compile it simply running _make_ (actually _make -j12_) and encountered the following error (w/ jemalloc).

[ 51%] Building CXX object tools/shell/linenoise/CMakeFiles/duckdb_linenoise.dir/linenoise.cpp.o
[ 51%] Building C object extension/jemalloc/jemalloc/CMakeFiles/jemalloc.dir/src/log.c.o
[ 51%] Building C object extension/jemalloc/jemalloc/CMakeFiles/jemalloc.dir/src/malloc_io.c.o
[ 52%] Building CXX object third_party/re2/CMakeFiles/duckdb_re2.dir/re2/simplify.cc.o
[ 52%] Building CXX object tools/shell/linenoise/CMakeFiles/duckdb_linenoise.dir/linenoise-c.cpp.o
/home/x/duckdb/extension/jemalloc/jemalloc/src/malloc_io.c: In function ‘duckdb_je_buferror’:
/home/x/duckdb/extension/jemalloc/jemalloc/src/malloc_io.c:107:16: error: returning ‘char *’ from a function with return type ‘int’ makes integer from pointer without a cast [-Wint-conversion]
  107 |         return strerror_r(err, buf, buflen);
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
gmake[3]: *** [extension/jemalloc/jemalloc/CMakeFiles/jemalloc.dir/build.make:552: extension/jemalloc/jemalloc/CMakeFiles/jemalloc.dir/src/malloc_io.c.o] Error 1
gmake[2]: *** [CMakeFiles/Makefile2:7952: extension/jemalloc/jemalloc/CMakeFiles/jemalloc.dir/all] Error 2
gmake[2]: *** Waiting for unfinished jobs....
[ 52%] Building CXX object tools/sqlite3_api_wrapper/sqlite3_udf_api/CMakeFiles/sqlite3_udf_api.dir/cast_sqlite.cpp.o
[ 52%] Building CXX object third_party/re2/CMakeFiles/duckdb_re2.dir/re2/stringpiece.cc.o
[ 52%] Building CXX object tools/shell/linenoise/CMakeFiles/duckdb_linenoise.dir/rendering.cpp.o
[ 52%] Building CXX object third_party/re2/CMakeFiles/duckdb_re2.dir/re2/tostring.cc.o
[ 52%] Building CXX object third_party/re2/CMakeFiles/duckdb_re2.dir/re2/unicode_casefold.cc.o
[ 52%] Building CXX object tools/shell/linenoise/CMakeFiles/duckdb_linenoise.dir/terminal.cpp.o
[ 52%] Building CXX object third_party/re2/CMakeFiles/duckdb_re2.dir/re2/unicode_groups.cc.o
[ 53%] Building CXX object third_party/re2/CMakeFiles/duckdb_re2.dir/util/rune.cc.o
[ 53%] Building CXX object third_party/re2/CMakeFiles/duckdb_re2.dir/util/strutil.cc.o
[ 53%] Built target duckdb_storage_serialization
[ 53%] Linking CXX static library libduckdb_re2.a
[ 53%] Built target duckdb_re2
[ 53%] Linking CXX static library libduckdb_fmt.a
[ 53%] Built target duckdb_fmt
[ 53%] Built target duckdb_linenoise
[ 53%] Built target duckdb_storage_table
[ 53%] Built target sqlite3_udf_api
[ 53%] Linking CXX static library libduckdb_fastpforlib.a
[ 53%] Built target duckdb_fastpforlib
gmake[1]: *** [Makefile:136: all] Error 2
gmake[1]: Leaving directory '/home/x/duckdb/build/release'
make: *** [Makefile:283: release] Error 2

**Comment by krlmlr** [comment]
Thanks. Can you please raise this with https://github.com/duckdb/duckdb ?

**Comment by xbasics** [comment]
Appreciate your reply!  Note that when compiling duckdb/duckdb v1.0.0 I don't encounter any errors.  It's only when I compile duckdb/duckdb-r.  So, you're saying that this nonetheless should be raised with them?

**Comment by vishalbelsare** [comment]
Facing same error trying to install duckdb in R under Arch Linux.

**Comment by krlmlr** [comment]
Thanks, Alex. I missed that part when you said that the upstream library compiles without error.

Is there a chance I can reproduce this locally with Docker?

**Comment by xbasics** [comment]
Sorry for the late reply.  Probably all you need is a Linux distro making sure you have the latest versions of GCC (14.1.0) and glibc (2.39).  I'm also using the latest version of R (v4.4.1).

Then again, you could also go with the latest version of Slackware and then updating it using the "current" channel...

https://alien.slackbook.org/blog/my-docker-packages-for-slackware-current/

Installing Slackware is really straight-forward.  In case of doubt, install everything and yet it won't be incredibly big.
According to [vishalbelsare], he's having the same issue with Arch Linux, so that's another avenue to try.  

There is really nothing special about my installation, other than Slackware-current, which runs very up-to-date versions of software.

**Comment by david-cortes** [comment]
This issue appears to have been an error in the RE2 library when AVX2 instructions are enabled:
https://github.com/duckdb/duckdb/pull/12344

It is fixed in the current `master` branch, so compilation succeeds if installing the R package for `duckdb` from GitHub, but fails when installing from CRAN.

Odd that the CRAN checks did not catch it, but would be ideal if this fix could be added to the CRAN 1.0 release, given that it might be a while until there's a next version release of core DuckDB.

**Comment by krlmlr** [comment]
Thanks for the heads-up. The change in 7353b83ffc683a6648cbd85d69a7b77ae69b996a would apply to the current CRAN version 1.0.0-2, but given that the next regular duckdb CRAN release will contain it, I'd rather not make the effort.

@david-cortes: Can you confirm that 7353b83ffc683a6648cbd85d69a7b77ae69b996a works but its predecessor fbbfe5addb2391b09720b3e235dfff07ea7ca740 fails?

**Comment by vishalbelsare** [comment]
> This issue appears to have been an error in the RE2 library when AVX2 instructions are enabled: [duckdb/duckdb#12344](https://github.com/duckdb/duckdb/pull/12344)
> 
> It is fixed in the current `master` branch, so compilation succeeds if installing the R package for `duckdb` from GitHub, but fails when installing from CRAN.
> 
> Odd that the CRAN checks did not catch it, but would be ideal if this fix could be added to the CRAN 1.0 release, given that it might be a while until there's a next version release of core DuckDB.

Yes, it worked for me when I installed from Github, so definitely odd that the CRAN had marked their build as OK.

**Comment by krlmlr** [comment]
Thanks, Vishal. Can you also confirm that 7353b83ffc683a6648cbd85d69a7b77ae69b996a works but its predecessor fbbfe5addb2391b09720b3e235dfff07ea7ca740 fails?

**Comment by david-cortes** [comment]
> Thanks, Alex. I missed that part when you said that the upstream library compiles without error.
> 
> Is there a chance I can reproduce this locally with Docker?

It's a matter of adding `-mavx2` or higher to the compilation arguments. For example, a file `~/.R/Makevars` like this would do:
```
CXXFLAGS += -mavx2
CXX11FLAGS += -mavx2
CFLAGS += -mavx2
```

**Comment by krlmlr** [comment]
Please open a new issue if this is still a problem.
